Library/SharedFunctions.swift

```diff
@@ -102,3 +104,46 @@ public func currencySymbol(forCountry country: Project.Country) -> String {
     return "\(String.nbsp)\(country.countryCode)\(country.currencySymbol)\(String.nbsp)"
   }
 }
+
+/// Returns a signal producer that emits, every second, the number of days/hours/minutes/seconds until 
+/// a date is reached, at which point it completes.
+///
+/// - parameter untilDate: The date to countdown to.
+///
+/// - returns: A signal producer.
+public func countdownProducer(to date: Date)
+  -> SignalProducer<(day: String, hour: String, minute: String, second: String), NoError> {
```

Comment:
this is a fun function! we use it in the countdown controller and the countdown of the live stream discovery cells.

lots of tests here: https://github.com/kickstarter/ios-oss/pull/53/files#diff-c7a229a8e1612db2ccad1308b69894a5